src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/LineGraphExport.java

```diff
@@ -91,52 +76,59 @@ public static AnalysisResult toLineGraphFormat(final CommitDiff commitDiff, fina
      * @param options Configuration options for the export, such as the format used for node and edge labels.
      * @return The number of the next diff tree to export (updated value of treeCounter).
      */
-    public static AnalysisResult toLineGraphFormat(final String repoName, final CommitDiff commitDiff, final StringBuilder lineGraph, final DiffTreeLineGraphExportOptions options) {
+    public static AnalysisResult toLineGraphFormat(final String repoName, final CommitDiff commitDiff, LineGraphExportOptions options, OutputStream destination) throws IOException {
         final AnalysisResult result = new AnalysisResult(repoName);
 
-        final String hash = commitDiff.getCommitHash();
         for (final PatchDiff patchDiff : commitDiff.getPatchDiffs()) {
-            if (patchDiff.isValid()) {
-                //Logger.info("  Exporting DiffTree #{}", treeCounter);
-                final Pair<DiffTreeSerializeDebugData, String> patchDiffLg;
-                try {
-                    patchDiffLg = toLineGraphFormat(patchDiff.getDiffTree(), options);
-                } catch (Exception e) {
-                    options.onError().accept(patchDiff, e);
-                    break;
-                }
-
-                if (patchDiffLg != null) {
-                    result.debugData.append(patchDiffLg.first());
-                    composeTreeInLineGraph(lineGraph, patchDiff, patchDiffLg.second(), options);
-                    ++result.exportedTrees;
-                }
-            } else {
-                Logger.debug("  Skipping invalid patch for file {} at commit {}", patchDiff.getFileName(), hash);
+            try {
+                toLineGraphFormat(patchDiff, options, destination, result);
+            } catch (Exception e) {
+                options.onError().accept(patchDiff, e);
+                break;
             }
         }
 
         result.exportedCommits = 1;
-        result.filterHits = new ExplainedFilterSummary(options.treeFilter());
 
         return result;
     }
```

Comment:
Wow this became so much easier! But I guess you forgot to compose the result of the `toLineGraphFormat` call with the `AnalysisResult result`? Previously all the debug data was collected in the result but now it isn't because the result is ignored, right?

Reply:
The overload that is called in that case does return `void` because it passes the `AnalysisResult` as parameter, but I don't remember why I did that.

I will change that too.